herring/puzzles/discordbot.py

```diff
@@ -223,12 +226,38 @@ async def on_raw_reaction_add(self, payload: discord.RawReactionActionEvent):
         # ignore myself
         if payload.user_id == self.bot.user.id:
             return
-        if payload.channel_id == self.announce_channel.id and payload.emoji.name == SIGNUP_EMOJI:
+        channel = self.bot.get_channel(payload.channel_id) or await self.bot.fetch_channel(payload.channel_id)
+        message: discord.Message = await channel.fetch_message(payload.message_id)
+        if payload.emoji.name == SIGNUP_EMOJI:
             # add someone to the puzzle
-            message: discord.Message = await self.announce_channel.fetch_message(payload.message_id)
-            await message.remove_reaction(SIGNUP_EMOJI, payload.member)
-            target_channel = message.channel_mentions[0]
-            await self.add_user_to_puzzle(payload.member, target_channel.name)
+            # we're ok with this working anywhere for any reason if anyone ever mentions a puzzle channel
+            if len(message.raw_channel_mentions) > 0:
```

Comment:
Does `raw_channel_mentions` in fact work for what we need here, for both regular and DM messages? (I saw in ryan's PR that regular `channel_mentions` did not.)

Reply:
Yes, it does. All `channel_mentions` does is map `message.guild.get_channel` over `raw_channel_mentions`, but if `message` is a DM, that doesn't work because it doesn't have a guild. So I'm doing it myself.